qiskit/dagcircuit/dagcircuit.py

```diff
@@ -564,19 +571,29 @@ def _map_condition(wire_map, condition, target_cregs):
         else:
             # if there is a condition, map the condition bits to the
             # composed cregs based on the wire_map
-            cond_creg = condition[0]
+            is_reg = False
+            if isinstance(condition[0], Clbit):
+                cond_creg = [condition[0]]
+            else:
+                cond_creg = condition[0]
+                is_reg = True
             cond_val = condition[1]
             new_cond_val = 0
             new_creg = None
             for bit in wire_map:
                 if bit in cond_creg:
-                    try:
-                        candidate_creg = next(creg
-                                              for creg in target_cregs
-                                              if wire_map[bit] in creg)
-                    except StopIteration as ex:
-                        raise DAGCircuitError('Did not find creg containing '
-                                              'mapped clbit in conditional.') from ex
+                    if is_reg:
+                        try:
+                            candidate_creg = next(creg
+                                                  for creg in target_cregs
+                                                  if wire_map[bit] in creg)
+                        except StopIteration as ex:
+                            raise DAGCircuitError('Did not find creg containing '
+                                                  'mapped clbit in conditional.') from ex
```

Comment:
A nice generalization that's enabled by this PR (and maybe something consider for a future issue/contribution) is to allow conditionals that span a list of arbitrary circuit `Clbit`s (even if they live in different `Register`s).  That would allow us to remove this `DAGCircuitError` (in the case where the condition doesn't map cleanly to a different register) and instead map the condition to a list of `Clbit`s.

Reply:
So we have been thinking about the generalization: conditioning on a list of ```Clbits```. I would certainly be happy to work on it.